zombies.js

```diff
@@ -78,137 +94,216 @@
+   * Return false in this case.
+   *
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+  discardItem(item) {
+    let selectItem = this.getPack().indexOf(item)
+    if (selectItem === -1) {
+      console.log("item not found")
+    } else {
+      this.getPack().splice(selectItem, 1)
+      console.log(selectItem + "-item")
+      return true;
+    }
+
+  }
+
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+  equip(itemToEquip) {
+    let selectItem = this.getPack().indexOf(itemToEquip)
+    if (itemToEquip instanceof Weapon && selectItem !== -1) {
+      if (this.equipped !== false) {
+        this.getPack().splice(itemToEquip, 1, this.equipped)
+        this.equipped = itemToEquip;
+      } else if (this.equipped === false)
+        this.equipped = itemToEquip;
+      this.discardItem(itemToEquip)
+    } else {
+      return false;
+    }
+  }
+
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+  eat(itemToEat) {
+    let eatItem = this.getPack().indexOf(itemToEat)
+    if (itemToEat instanceof Food && eatItem !== -1) {
+      if (itemToEat.energy + this.health > this.getMaxHealth()) {
+        this.health = this.getMaxHealth();
+        this.discardItem(itemToEat);
+      } else if (itemToEat.energy + this.health <= this.getMaxHealth())
+        this.health += itemToEat.energy;
+      this.discardItem(itemToEat);
```

Comment:
formatting.

You could optimize this area a bit by only calling `this.discardItme(itemToEat)` once instead of in both the if and the else if.